src/jit/morph.cpp

```diff
@@ -9550,8 +9550,7 @@ GenTreePtr          Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* ma
         {
             GenTreePtr effOp1 = op1->gtEffectiveVal();
             
-            noway_assert( (effOp1->gtOper == GT_CNS_INT) &&
-                         ((effOp1->gtIntCon.gtIconVal == 0) || (effOp1->gtIntCon.gtIconVal == 1)) );
+            noway_assert(effOp1->IsZero() || effOp1->IsIOne());
```

Comment:
> I don't think these contexts are at all difficult to understand

They may be for new readers. I personally would feel more comfortable if the assumptions being made at these various checks were covered by assertions.